pkg/resource/resid.go

```diff
@@ -97,10 +97,35 @@ func (n ResId) Namespace() string {
 
 // CopyWithNewPrefix make a new copy from current ResId and append a new prefix
 func (n ResId) CopyWithNewPrefix(p string) ResId {
-	return ResId{gvk: n.gvk, name: n.name, prefix: p + n.prefix, namespace: n.namespace}
+	return ResId{gvk: n.gvk, name: n.name, prefix: n.concatePrefix(p), namespace: n.namespace}
 }
 
 // CopyWithNewNamespace make a new copy from current ResId and set a new namespace
 func (n ResId) CopyWithNewNamespace(ns string) ResId {
 	return ResId{gvk: n.gvk, name: n.name, prefix: n.prefix, namespace: ns}
 }
+
+// HasSamePrefix check if two ResIds have the same leading prefix
```

Comment:
This comment suggests that this is some kind of equals method.  It's much odder than that.

Please elaborate on the empty string checks and the check of only the first element in the comment, especially since this is a public method.

Perhaps rename to HasSameEffectivePrefix ?

Reply:
Naming is always hard. `HasSameEffectivePrefix` sounds better than `HasSamePrefix`. 
 I'll come up with a good comment of this function.

Reply:
I found that we could get rid the special case for empty prefix. The new change looks more clean.